OpenRA.Mods.RA/Traits/Disguise.cs

```diff
@@ -48,10 +48,16 @@ public Player Owner
 		{
 			get
 			{
-				if (disguise.Disguised)
-					return self.Owner == self.World.LocalPlayer ? self.Owner : disguise.AsPlayer;
+				if (!disguise.Disguised)
+					return self.Owner;
 
-				return self.Owner;
+				Player p = null;
+				if (self.World.LocalPlayer != null && self.World.LocalPlayer.WinState == WinState.Undefined)
```

Comment:
With player with disabled shroud cheat you mean allied player, right? Because enemy player should not see through disguise even with disable shroud cheat.

Also, ```RenderPlayer.InternalName == "Everyone" ||``` is not necessary since that is already allied with all players, the second part would cover it.